src/shared/views/BuildingFootprint/Validator.js

```diff
@@ -141,76 +164,135 @@ class _Validator extends React.Component<Props, State> {
         } = this.props;
         const { currentTaskIndex } = this.state;
         if (result !== null && result !== undefined) {
+            // the user tapped a button, save result and update how far
+            // we've worked
             submitResult(result, this.expandedTasks[currentTaskIndex].taskId);
-            this.tasksDone = currentTaskIndex;
+            if (currentTaskIndex > this.tasksDone) {
+                this.tasksDone = currentTaskIndex;
+            }
         } else if (currentTaskIndex > this.tasksDone) {
```

Comment:
I don't get this line. it looks exactly the same as you have in line 170

Reply:
Correct. But the one above is under another condition. The first one is when the user taps a button, the second one when they swipe.